openshift_tools/monitoring/generic_metric_sender.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# vim: expandtab:tabstop=4:shiftwidth=4
+"""
+Base class for sending metrics
+"""
+
+import os
+import yaml
+
+
+class GenericMetricSenderException(Exception):
+    '''
+        GenericMetricSenderException
+        Exists to propagate errors up from the api
+    '''
+    pass
+
+
+class GenericMetricSender(object):
+    """
+    Abstract class that encapsulates all metric sending operations.
+    """
+
+    def __init__(self):
+        self.config = None
+        self.config_file = None
+
+    def parse_config(self):
+        """ parse default config file """
+        if not self.config:
+            if not os.path.exists(self.config_file):
+                raise GenericMetricSenderException(self.config_file + " does not exist.")
+            self.config = yaml.load(file(self.config_file))
+
+    def add_dynamic_metric(self, discovery_key, macro_string, macro_array, host=None, synthetic=False):
+        pass
```

Comment:
Ok let's take https://github.com/openshift/openshift-tools/blob/stg/scripts/monitoring/cron-send-disk-metrics.py#L70 for example. So you want to replace this call with add_dynamic_metric. but do we want to add anther API for adding tags, and change the script to call it or do we want the hawk_sender to infer tags from the parameters sent to add_dynamic_metric?